go/fn/run.go

```diff
@@ -31,10 +31,8 @@ func AsMain(input interface{}) error {
 	err := func() error {
 		var p ResourceListProcessor
 		switch input := input.(type) {
-		case Runner:
-			p = runnerProcessor{fnRunner: input}
-		case *Runner:
-			p = runnerProcessor{fnRunner: *input}
+		case runnerProcessor:
+			p = input
 		case ResourceListProcessorFunc:
```

Comment:
nit: suggest combining the two cases

```
case runnerProcessor, ResourceListProcessorFunc:
  p = input
```

I'm also wondering if the switch statement still makes sense here, or would it be simpler now to just check the type of input with an if statement?

Reply:
I think we cannot combining the two types if you want a dedicated type assertion. In that case, the condition is valid but the typed `input` would be read as `interface{}`.
  
`switch input := input.(type)` might make it a little bit inobvious, but basically the extended code is 
```go
switch input.(type) {
 case runnerProcessor:
       p = input.(runnerProcessor)
 case  ResourceListProcessorFunc:
       p = input.(ResourceListProcessorFunc)
}
```

Reply:
Ah I see, your current code makes more sense then. Thanks for the explanation!